src/Aspire.Dashboard/Components/Routes.razor

```diff
@@ -1,15 +1,31 @@
-﻿@using Microsoft.AspNetCore.Components.Authorization
+﻿@using Aspire.Dashboard.Model
+@using Aspire.Dashboard.Components.Resize
 @inject IStringLocalizer<Resources.Routes> Loc
 
-<Router AppAssembly="@typeof(App).Assembly">
-    <Found Context="routeData">
-        <!-- AuthorizeRouteView protects pages from being accessed when authorization is required -->
-        <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(Layout.MainLayout)" />
-    </Found>
-    <NotFound>
-        <PageTitle>@Loc[nameof(Resources.Routes.NotFoundPageTitle)]</PageTitle>
-        <LayoutView Layout="@typeof(Layout.MainLayout)">
-            <Aspire.Dashboard.Components.Pages.NotFound />
-        </LayoutView>
-    </NotFound>
-</Router>
+<BrowserDimensionWatcher @bind-ViewportInformation="@_viewportInformation" />
+
+@if (_viewportInformation is null)
+{
+    // prevent render until we've determined the browser viewport so that we don't have to re-render
+    // if we guess wrong
+    return;
+}
+
+<CascadingValue Value="@_viewportInformation">
```

Comment:
What happens when this changes? Does it cause everything to re-render? I noticed weird behavior on the console logs page with everything reseting on resize. We should be certain about what is happening.

Right now, I believe the value changes for every pixel resize. `ViewportInformation` has height and width, and then IsDesktop and IsUltraLowHeight are calculated from the dimensions.

I think this value should only change if either of these properties changes. That means the type only has IsDesktop and IsUltraLowHeight properties on it that are calculated when it is created. The value only changes if either of these properties changes.

Does that make sense?

Reply:
>  Right now, I believe the viewport value changes for every pixel resize. 
> That's pretty heavy weight overhead

Yep, you're right.

> I think this value should only change if either of these properties changes.

I agree with you, and have updated the type to only contain those two properties, as we don't really need width or height